### ISSUE TITLE
Missing collaborators

### ISSUE THREAD

**Issue by emanueleanselmo** [Yooooomi/your_spotify#444]
**Describe the bug**

When two or more artists collaborate, only the first artist (on the album cover, I think) is counted in the artist section, even though both are listed as collaborators in the album section.

**Expected behavior**

Both artists appear as collaborators and therefore have their stats, as seen in the album section.

**Additional context**

I'm talking about albums actually composed by both artists (e.g. Ambient 2: The Plateaux of Mirror by Harold Budd and Brian Eno). Here tracks appear as Budd's only

**Screenshots**

![{4CA4DC58-F561-47D7-BD4A-3647CB59E637}](https://github.com/user-attachments/assets/78901dbe-ae73-4daf-bc47-4a95e784c695)
![{41BFF0B1-28ED-4717-AF74-F7539E379A19}](https://github.com/user-attachments/assets/c7068be0-6cd0-4f7a-9c4b-9a02cdf3584f)

**Comment by M2K3K5** [comment]
Yes, you are right. I quickly checked and came to the same conclusion.

**Comment by emanueleanselmo** [comment]
Thanks for the answer and the app. I've never worked with Spotify API, but
reading the json I see that contributors are not really included by them,
since "master_metadata_album_artist_name" has a unique name as value. I
don't know if it's possible to create an array of names for this key, or
even if it's possible to create another key (e.g.
"master_metadata_album_collaborator_name"), but I don't want to try to do
it manually... also because when I import a file into the app, I'm no
longer able to delete it. What I do know is that another app stats.fm was
able to contain this problem, but at the same time it has other much more
serious bugs. I don't know if these things are related.

Il giorno mer 6 nov 2024 alle ore 08:51 M2K3K5 ***@***.***>
ha scritto:

**Comment by RomainNeup** [comment]
Hello @emanueleanselmo 👋 
Indeed, it will be hard to fix this. It's working when Spotify is providing us the info in the right field. 😕
